File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var secondHighest = function (s) {
    let nums = [];
    for (let i = 0; i < s.length; i++) {
        if (Number(s[i]) >= 0 || Number(s[i]) <= 9) {
            nums.push(Number(s[i]));
        }
    }
    
    const numss = [...new Set(nums)];
    if (numss.length <= 1) {
        return -1;
    }
    numss.sort((a, b) => b-a);
    console.log(nums);
    return numss[1];
};

File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.java
class Solution {
    public int secondHighest(String s) {
             List<Integer> nums = new ArrayList<>();

        
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (Character.isDigit(ch)) {
                nums.add(ch - '0'); 
            }
        }

        Set<Integer> uniqueSet = new HashSet<>(nums);

        if (uniqueSet.size() <= 1) {
            return -1;
        }

        List<Integer> uniqueList = new ArrayList<>(uniqueSet);
        uniqueList.sort(Collections.reverseOrder());

        
        return uniqueList.get(1);
   

    }
}